### PR TITLE
Publish metric on distribution availability

### PR DESCRIPTION
**Pull request by bhautikpip**
*Issue #, if available:*

*Description of changes:*

Publish metric on distribution availability

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/continuous-monitoring.yml [hunk 1]
```diff
@@ -11,9 +11,22 @@ jobs:
...
       - name: Run distribution test
+        id: distribution-availability
         run: go test distributioncheck
         working-directory: integration-tests/distributioncheck
+      
+      - name: Publish metric on X-Ray Go SDK distribution availability
+        if: ${{ always() }}
+        run: |
+          if [[ "${{ steps.distribution-availability.outcome }}" == "failure" ]]; then
+            aws cloudwatch put-metric-data --metric-name XRayGoSDKDistributionAvailability --dimensions failure=rate --namespace MonitorSDK --value 1 --timestamp $(date +%s)
```
Comment (willarmiros) [inline .github/workflows/continuous-monitoring.yml:28]: How do we know the AWS CLI is installed?

**Review by willarmiros — APPROVED** [review]
Just 1 question

File: .github/workflows/continuous-monitoring.yml (hunk 1 above)
Comment (bhautikpip) [reply .github/workflows/continuous-monitoring.yml:28]: Github instance installs the latest version of AWS CLI by default.

Comment (anuraaga) [reply .github/workflows/continuous-monitoring.yml:28]: Don't we need to add aws login? Also what account are we going to use? Ideally we use https://github.com/aws-observability/aws-otel-ops to set up the secrets.

Comment (bhautikpip) [reply .github/workflows/continuous-monitoring.yml:28]: we need to configure aws credentials on Github instance. We are using the testing account for this so we can use secrets that are set up for integration test workflow no need to set up another secrets.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**